Some updates from upstream to handle more/newer cpu types

--- beecrypt-4.1.2/acinclude.m4
+++ beecrypt-4.1.2/acinclude.m4
@@ -16,7 +16,7 @@
   i[[3456]]86)
     case $withval in
     i[[3456]]86 | \
-    pentium | pentium-mmx | pentiumpro | pentium[[234]] | \
+    pentium | pentium-m | pentium-mmx | pentiumpro | pentium[[234]] | \
     athlon | athlon-tbird | athlon-4 | athlon-xp | athlon-mp)
       ;;
     *)
@@ -80,9 +80,10 @@
   case $target_cpu in
   i[[3456]]86)
     case $withval in
+    em64t | \
     i[[3456]]86 | \
-    pentium | pentium-mmx | pentiumpro | pentium[[234]] | \
-    athlon | athlon-tbird | athlon-4 | athlon-xp | athlon-mp)
+    pentium | pentium-m | pentium-mmx | pentiumpro | pentium[[234]] | \
+    athlon | athlon-tbird | athlon-4 | athlon-xp | athlon-mp | athlon64 | k8)
       if test "$ac_with_cpu" != yes; then
         bc_target_cpu=$withval
       fi
--- beecrypt-4.1.2/configure.ac
+++ beecrypt-4.1.2/configure.ac
@@ -502,9 +513,11 @@
   arm*)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_ARM"
     ;;
+  x86_64 | athlon64 | athlon-fx | em64t | k8)
+    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_X86_64"
+    ;;
   athlon*)
-    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I386 -DOPTIMIZE_MMX"
-    CFLAGS="$CFLAGS -mmmx"
+    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX"
     ;;
   i386)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I386"
@@ -526,24 +536,23 @@
   pentium)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I586"
     ;;
+  pentium-m)
+    CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX -DOPTIMIZE_SSE -DOPTIMIZE_SSE2"
+    ;;
   pentium-mmx)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I586 -DOPTIMIZE_MMX"
-    CFLAGS="$CFLAGS -mmmx"
     ;;
   pentiumpro)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686"
     ;;
   pentium2)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX"
-    CFLAGS="$CFLAGS -mmmx"
     ;;
   pentium3)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX -DOPTIMIZE_SSE"
-    CFLAGS="$CFLAGS -msse"
     ;;
   pentium4)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_I686 -DOPTIMIZE_MMX -DOPTIMIZE_SSE -DOPTIMIZE_SSE2"
-    CFLAGS="$CFLAGS -msse2"
     ;;
   powerpc)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_POWERPC"
@@ -556,15 +565,12 @@
     ;;
   sparcv8)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_SPARCV8"
-    CFLAGS="$CFLAGS -Wa,-xarch=v8"
     ;;
   sparcv8plus*)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_SPARCV8PLUS"
-    CFLAGS="$CFLAGS -Wa,-xarch=v8plus"
     ;;
   sparcv9*)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_SPARCV9"
-    CFLAGS="$CFLAGS -Wa,-xarch=v9"
     ;;
   x86_64)
     CPPFLAGS="$CPPFLAGS -DOPTIMIZE_X86_64"
